Improve http client docs (#994)
SAP/cloud-sdk: docs-js/features/connectivity/generic-http-client.mdx

The HTTP client provided by SAP Cloud SDK allows to automatically retrieve a CSRF token for HTTP APIs which support this.
Fetching CSRF token is automatically enabled for all requests that typically might change data, such as `PUT`, `POST`, `DELETE` or `PATCH` requests, and is not available for `GET` requests.

> Fetching the CSRF token can be enabled or disabled via the `httpRequestOptions` parameter of `executeHttpRequest`:

Reviewer: ```suggestion
Set `httpRequestOptions` parameter of `executeHttpRequest` to `false` to disable the CSRF token fetching:
```

Author: No this is not correct because the `fetchCsrfToken` field of the `httpRequestOptions` parameter needs to be set to `false`.

```ts
{

>   fetchCsrfToken: BOOLEAN_TRUE_OR_FALSE;

Reviewer: [pp] I think this code block is unnecessary and can be removed since the lines above and below are clear enough.

Author: I think this helps with clarity and I'd like to leave it in.

}
```

> For example, to avoid CSRF token fetching when doing a `PUT`, `POST`, `DELETE` or `PATCH` request, disable it like in this example:

Reviewer: [pp] Also remove this line as it has duplicated meaning if we adopt the suggestion.

Author: I think this helps with clarity and I'd like to leave it in.

```ts
const response = await executeHttpRequest(
  {

>     url: `https://httpbin.org/post`

Reviewer: [pp] I would avoid using a 3rd party website here.

Author: I see your point, but my reasoning is that this allows us to have actually working examples which users can copy/paste, and wich we can have working in tests in the samples repo. Unless there are any legal issues with that (which I'm not aware of) I'd like to keep it.

  },
  {
    method: 'post'
  },
  {
    fetchCsrfToken: false
  }
);
```

> In this example, we can only do the request when `fetchCsrfToken` is set to `false` because the httpbin endpoint does not implement CSRF token fetching.

Reviewer: ```suggestion
Disable the CSRF token fetching if the endpoint does not support it.
```
Make it more general (or perhaps remove it since people should probably already know if csrf is supported.)

Author: I think this sentence explains what's happening in the example and it adds value imo, so I'd like to leave it.

> For an example with CSRF token fetching enabled see the [http-client.spec.ts](https://github.com/SAP-samples/cloud-sdk-js/blob/main/samples/http-client-examples/http-client.spec.ts) example.

Reviewer: [req] Reformulate the sentence to avoid using the word `examples` too often.

  for (const k in params) {
    // Customize your required encoding logic here
    encodedParams[k] = params[k].toString().replace('x', 'y');
  }

  return encodedParams;
};

const response = await executeHttpRequest(
  {

>     url: 'https://httpbin.org/anything'

Reviewer: Same as above. Avoid using 3rd party URL. Use `example.com` instead as it is [reserved from IANA](https://en.wikipedia.org/wiki/Example.com).

In this example, the `custom` `apiKey` and `myParam` override the request config:

```ts
const response = await executeHttpRequestWithOrigin(
  {

>     url: 'https://httpbin.org/anything'

Reviewer: Same as above.
